### PR TITLE
Fixes 20191115

### PR DESCRIPTION
**Pull request by lyakh**
These are (supposedly) fixes / improvements from #1269 - please, have a look, if needed, I can also submit them one by one. None of them is critical, just minor issues.

### CODE REVIEW

File: sound/soc/sof/intel/hda-ipc.c [hunk 1]
```diff
@@ -194,33 +222,8 @@ irqreturn_t hda_dsp_ipc_irq_thread(int irq, void *context)
...
-			/* normal message - process normally */
-			snd_sof_ipc_msgs_rx(sdev);
-		}
-
-		hda_dsp_ipc_host_done(sdev);
-
-		ipc_irq = true;
-	}
+	if (host_msg)
+		snd_sof_ipc_msgs_rx(sdev);
```
Comment (plbossart) [inline sound/soc/sof/intel/hda-ipc.c:226]: this changes the flow for a nornal message between snd_sof_ipc_msgs_rx(sdev) and hda_dsp_ipc_host_done(sdev);

Is this needed? desirable?

File: sound/soc/sof/topology.c [hunk 2]
```diff
@@ -1437,7 +1437,7 @@ static int sof_widget_load_pcm(struct snd_soc_component *scomp, int index,
 	ret = sof_ipc_tx_message(sdev->ipc, host->comp.hdr.cmd, host,
 				 sizeof(*host), r, sizeof(*r));
 	if (ret >= 0)
-		return ret;
+		return 0;
```
Comment (plbossart) [inline sound/soc/sof/topology.c:1440]: can ret be > 0?
If yes, there is an inconsistency with your previous patch.
if no, then the test should be changed as well?

File: sound/soc/sof/topology.c [hunk 3]
```diff
@@ -3330,12 +3329,9 @@ int snd_sof_complete_pipeline(struct snd_sof_dev *sdev,
...
 
-	ret = sof_ipc_tx_message(sdev->ipc,
-				 ready.hdr.cmd, &ready, sizeof(ready), &reply,
-				 sizeof(reply));
-	if (ret < 0)
-		return ret;
-	return 1;
+	return !sof_ipc_tx_message(sdev->ipc,
+				   ready.hdr.cmd, &ready, sizeof(ready), &reply,
+				   sizeof(reply));
```
Comment (plbossart) [inline sound/soc/sof/topology.c:3334]: if res can be >0 then this is wrong.

File: sound/soc/sof/intel/hda-ipc.c [hunk 4]
```diff
@@ -148,6 +149,33 @@ irqreturn_t hda_dsp_ipc_irq_thread(int irq, void *context)
 	hipci = snd_sof_dsp_read(sdev, HDA_DSP_BAR, HDA_DSP_REG_HIPCI);
 	hipcte = snd_sof_dsp_read(sdev, HDA_DSP_BAR, HDA_DSP_REG_HIPCTE);
 
+	/* is this a new message from DSP */
```
Comment (ranj063) [inline sound/soc/sof/intel/hda-ipc.c:152]: @lyakh I can only envision this possibility if the FW panic happens while the host is processing the reply in which case we will go on and process the panic right after. So not sure if this change is absolutely required.

File: sound/soc/sof/topology.c (hunk 3 above)
Comment (ranj063) [reply sound/soc/sof/topology.c:3334]: @lyakh the logic for this is completely broken I agree. I think we should just remove swidget->complete. It is unlikely that the widget is added to the list more than once.

If this logic did work, we'd never be able to do complete_pipeline after a suspend/resume cycle. So thank god for that :)

File: sound/soc/sof/topology.c (hunk 2 above)
Comment (ranj063) [reply sound/soc/sof/topology.c:1440]: @lyakh @plbossart there's no guarantee that the sof_ipc_tx_message() cannot be > 0. It depends on whatever the FW reply contains. But I think the "right" fix would be to change sof_ipc_tx_message() to return 0 if the reply return value is >= 0

**Comment by plbossart** [comment]
@lyakh no replies in 21 days, if this PR is abandoned then please close it, if not please provide an update.

**Pull request closed without merging**